src/main/java/com/redhat/rhjmc/containerjfr/net/SslConfiguration.java

```diff
@@ -0,0 +1,207 @@
+            throw new IllegalArgumentException("both KEY_PATH and CERT_PATH must be specified");
+        }
+
+        Path key = Path.of(env.getEnv(KEY_PATH_ENV)).normalize();
+        Path cert = Path.of(env.getEnv(CERT_PATH_ENV)).normalize();
+        if (!fs.exists(key)) {
+            throw new IllegalArgumentException("KEY_PATH refers to a non-existent file");
+        }
+
+        if (!fs.exists(cert)) {
+            throw new IllegalArgumentException("CERT_PATH refers to a non-existent file");
+        }
+
+        return new Pair<>(key, cert);
+    }
+
+    Path discoverKeyStorePathInDefaultLocations() {
+        Path home = Path.of(System.getProperty("user.home", "/"));
+
+        if (fs.exists(home.resolve("container-jfr-keystore.jks"))) {
+            return home.resolve("container-jfr-keystore.jks");
+        } else if (fs.exists(home.resolve("container-jfr-keystore.pfx"))) {
+            return home.resolve("container-jfr-keystore.pfx");
+        } else if (fs.exists(home.resolve("container-jfr-keystore.p12"))) {
+            return home.resolve("container-jfr-keystore.p12");
+        }
+
+        return null;
+    }
+
+    Pair<Path, Path> discoverKeyCertPathPairInDefaultLocations() {
+        Path home = Path.of(System.getProperty("user.home", "/"));
+        Pair<Path, Path> ret = new Pair<>(null, null);
+
+        // discover keyPath
+        if (fs.exists(home.resolve("container-jfr-key.pem"))) {
+            ret.left = home.resolve("container-jfr-key.pem");
+        } else {
+            logger.warn("unable to locate container-jfr-key.pem");
+        }
+
+        // discover certPath
+        if (fs.exists(home.resolve("container-jfr-cert.pem"))) {
+            ret.right = home.resolve("container-jfr-cert.pem");
+        } else {
+            logger.warn("unable to locate container-jfr-cert.pem");
+        }
+
+        if (keyPath == null || certPath == null) {
+            return null;
+        }
+
+        return ret;
+    }
+
+    HttpServerOptions setToHttpServerOptions(HttpServerOptions options) {
+        if (!enabled()) {
+            return options.setSsl(false);
+        }
+
+        options.setSsl(true);
+
+        if (keystorePath != null) {
+            if (keystorePass.isEmpty()) {
+                logger.warn("keystore password unset or empty");
+            }
+
+            if (keystorePath.endsWith(".jks")) {
+                return options
+                        .setKeyStoreOptions(new JksOptions().setPath(keystorePath.toString()).setPassword(keystorePass));
+            } else if (keystorePath.endsWith(".pfx") || keystorePath.endsWith(".p12")) {
+                return options
+                        .setPfxKeyCertOptions(new PfxOptions().setPath(keystorePath.toString()).setPassword(keystorePass));
+            }
+
+            IllegalArgumentException e = new IllegalArgumentException("unrecognized keystore type");
+            logger.error(e);
+            throw e;
```

Comment:
Creating a runtime exception, logging it, and then throwing it seems weird. IMO this should be a checked exception and whoever is responsible for catching it should then also deal with logging it. And `IllegalArgumentException` generally seems like not quite the right fit either, since it's being thrown not based on the arguments passed to the method throwing the exception but based on some previously set internal state of the object. For existing runtime exceptions I can think of, IllegalStateException sounds more applicable.

Maybe this warrants a new, checked, `SslConfigurationException` for this purpose.

Reply:
I agree. A new checked `SslConfigurationException` sounds desirable. I'll create and switch to that instead.